src/crappy/camera/meta_camera/camera_setting/camera_scale_setting.py

```diff
@@ -37,19 +38,46 @@ def __init__(self,
       getter: The method for getting the current value of the setting.
       setter: The method for setting the current value of the setting.
       default: The default value to assign to the setting.
+      step: The step value for the variation of the setting values.
     """
 
     self.lowest = lowest
     self.highest = highest
-    self.type = int if isinstance(lowest + highest, int) else float
+    self.step = step
+    self.type = int if isinstance(self.lowest + self.highest, int) else float
 
     if default is None:
-      default = self.type((lowest + highest) / 2)
+      default = self.type((self.lowest + self.highest) / 2)
     else:
       default = self.type(default)
 
     super().__init__(name, getter, setter, default)
 
+    if self.step:
+      if self.type == int and isinstance(self.step, float):
+        self.step = int(self.step)
+        self.log(logging.WARNING, f"Could not set {self.name} steps to "
+                                  f"{step} (lowest: int, step: float), "
+                                  f"the step is now {self.step} !")
+      if self.step > (self.highest - self.lowest):
+        self.step = 1 if self.type == int else (self.highest -
+                                                self.lowest) / 1000
+        self.log(logging.WARNING, f"Could not set {self.name} steps to "
+                                  f"{step}, the step is now {self.step} !")
+      if (self.highest - self.lowest) % self.step:
```

Comment:
Should only be checked if `self.type` is `int`.
If you run `1.0 % 0.1` in a Python console you get `0.99999...`, which is not zero !